cfn-resources/datalakes/cmd/resource/resource.go

```diff
@@ -410,24 +394,20 @@ func validateProgress(client *mongodbatlas.Client, currentModel *Model, targetSt
 	return p, nil
 }
 
-// function to check if record already exist
-func isExist(currentModel *Model, client *mongodbatlas.Client) bool {
+func isExist(currentModel *Model, client *util.MongoDBClient) bool {
 	projectID := *currentModel.ProjectId
 	tenantName := *currentModel.TenantName
-	dataLake, _, err := client.DataLakes.Get(context.Background(), projectID, tenantName)
-	if err != nil {
+	dataLake, _, err := client.AtlasV2.DataFederationApi.GetFederatedDatabase(context.Background(), projectID, tenantName).Execute()
+	if err != nil || dataLake == nil {
 		return false
 	}
-	if dataLake != nil {
-		return true
-	}
 
-	return false
+	return true
```

Comment:
this is really:   return err == nil && dataLake != nil

Reply:
updated